internal/utils/utils.go

```diff
@@ -15,3 +17,54 @@ func IsArgoCDCluster(labels map[string]string) bool {
 func SortSecretsByName(secrets *corev1.SecretList) {
 	sort.SliceStable(secrets.Items, func(i, j int) bool { return secrets.Items[i].Name < secrets.Items[j].Name })
 }
+
+// SortAppsByName sort the Application slice in place by the app name
+func SortAppsByName(apps *[]argov1alpha1.Application) {
+	sort.SliceStable(*apps, func(i, j int) bool { return (*apps)[i].Name < (*apps)[j].Name })
+}
+
+// GetAppsBySyncStatusCode returns the Applications matching the specified sync status code
+func GetAppsBySyncStatusCode(apps []argov1alpha1.Application, code argov1alpha1.SyncStatusCode) []argov1alpha1.Application {
+	var result []argov1alpha1.Application
+
+	for _, app := range apps {
+		if app.Status.Sync.Status == code {
+			result = append(result, app)
+		}
+	}
+
+	SortAppsByName(&result)
```

Comment:
Is this required?

Reply:
I think so, because then we don't rely on the client implementation and when we write the integration test we have a predictable output.
Do you see it as an issue?

Reply:
I'd move the sorting in the test, as you only want the sorting to validate your testing if that makes sense

Reply:
Ah, I haven't thought about that. I was considering that more as a precursor of https://github.com/Skyscanner/argocd-progressive-rollout/issues/9

Let me see how easy is to remove it.

Reply:
Fixed